internal/pkg/configloader/config.go

```diff
@@ -47,14 +52,24 @@ func LoadConf(fname string) *Config {
 		log.Fatal("See more help by running \"dtm help\"")
 	}
 
-	config := Config{}
-
+	var config Config
 	err = yaml.Unmarshal(fileBytes, &config)
 	if err != nil {
-		log.Fatal(err)
+		log.Printf("unmarshal config failed: %s", err)
+		return nil
+	}
+
+	errs := config.Validate()
+	// valid config
+	if len(errs) == 0 {
+		return &config
 	}
 
-	return &config
+	// invalid config
+	for _, e := range errs {
```

Comment:
I think it's better to return &config as the last line of this function; it's easier for readers to get the whole picture without reading every line of the code. In the current structure, the reader has to dig to find out what is the normal return result.

Reply:
solved